src/str/normalize.rs

```diff
@@ -0,0 +1,28 @@
+use std::fmt;
+use Predicate;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NormalizedTrimPredicate<P>
```

Comment:
`Trim` is a word specific to removing whitespace.  This isn't doing that right?  So maybe just `NormalizedPredicate`?